twenty-first/src/math/tip5.rs

```diff
@@ -673,10 +675,39 @@ impl Sponge for Tip5 {
     }
 }
 
+impl Hasher for Tip5 {
+    fn finish(&self) -> u64 {
+        self.state[0].value()
+    }
+
+    fn write(&mut self, bytes: &[u8]) {
+        let bfield_elements = bytes.chunks(BFieldElement::BYTES).map(|chunk| {
+            let mut buffer = [0u8; BFieldElement::BYTES];
+            buffer[..chunk.len()].copy_from_slice(chunk);
+            BFieldElement::new(u64::from_le_bytes(buffer))
+        });
+
+        let binding: Vec<BFieldElement> = bfield_elements.collect();
+        let bfield_absorb_chunks = binding.chunks(Tip5::RATE);
+
+        bfield_absorb_chunks.for_each(|absorb_rate_chunk| {
+            let mut asborb_chunk_buffer: [BFieldElement; Tip5::RATE] =
+                [BFieldElement::new(0); Tip5::RATE];
```

Comment:
I'm a big fan of type inference and try to let it do whatever it can. How about:

```suggestion
            let mut asborb_chunk_buffer = [BFieldElement::ZERO; Tip5::RATE];
```

The type annotation aside, I personally like to use `BFieldElement::ZERO` over `BFieldElement::new(0)` because it is a constant. On the other hand `BFieldElement::new` is a `const fn` and the compiler might produce the same code in both cases. Both parts of the suggestion – removing the type annotation and using `ZERO` over `new` – are a matter of preference. Feel free to leave things as they are if you prefer that.